webxr-api/session.rs

```diff
@@ -238,12 +240,14 @@ where
                         swap_chain.recycle_surface(surface);
                     }
                 }
-                self.current_frame = self.device.wait_for_animation_frame();
-
-                if self.current_frame.is_none() {
-                    warn!("Device stopped providing frames, exiting");
-                    return false;
-                }
+                let frame = match self.device.wait_for_animation_frame() {
+                    Some(frame) => frame,
+                    None => {
+                        warn!("Device stopped providing frames, exiting");
+                        return false;
+                    }
+                };
+                let _ = self.frame_sender.send(frame);
```

Comment:
So if I understand correctly, this is the big difference here - immediately after finishing a frame we wait for the new frame, but instead of storing it in `current_frame` and then grabbing it next time script tells us that it's time to start a rAF, we send it to script immediately which then causes a rAF callback task to be queued.

Reply:
Yes. Note that in the current implementation script _always_ calls `request_animation_frame` immediately after `render_animation_frame`, so it seems like this should not cause any improvement, however it's possible that the extra IPC overhead plus the fact that the event loop is being pumped by the main thread which may decide to do something else are conspiring to cause problems.